### PR TITLE
Persist results to CrateDB for `tracking_merlion.py`

### PR DESCRIPTION
**Pull request by hammerhead**
Store known anomalies in a database table, instead of hard-coding them in Python. Then, save detected anomalies alongside in that table.

### CODE REVIEW

File: examples/tracking_merlion.py [hunk 1]
```diff
@@ -29,6 +29,7 @@
 """
 
 import os
+from datetime import timedelta
```
Comment (amotl) [inline examples/tracking_merlion.py:32]: It is similar like `import pandas as pd`.
```suggestion
import datetime as dt
```

File: examples/tracking_merlion.py [hunk 2]
```diff
@@ -113,14 +137,51 @@ def read_data(table_name: str) -> pd.DataFrame:
...
+def save_anomalies(table_name: str, anomalies: pd.DataFrame, mttd: int):
+    conn = connect_database()
+    with conn:
+        cursor = conn.cursor()
+        rows = []
+        for index, row in anomalies.iterrows():
+            rows.append(
+                (
+                    index,
+                    index + timedelta(seconds=mttd),
```
Comment (amotl) [inline examples/tracking_merlion.py:158]:
```suggestion
                    index + dt.timedelta(seconds=mttd),
```

File: tests/test_examples.py [hunk 3]
```diff
@@ -103,3 +103,28 @@ def test_tracking_merlion(reset_database, engine: sa.Engine, tracking_store: Sql
...
+                WHERE experiment_id = (
+                    SELECT experiment_id
+                    FROM testdrive.runs
+                    ORDER BY start_time DESC
+                    LIMIT 1
+                )
+                """
+            ),
+        )
+        assert anomalies_generated.rowcount >= 1
```
Comment (hammerhead) [inline tests/test_examples.py:130]: I'm getting varying results here. Generates three anomalies locally, but only one here on GitHub. Not sure why, but it shouldn't matter for testing the code path of getting the anomalies saved into the database.

**Comment by hammerhead** [comment]
There is a test failure for `tests.test_tracking.TestSqlAlchemyStore testMethod=test_log_inputs_with_duplicates_in_single_request` that also appears after a retry. Sounds unrelated to the example change? Do you have an idea where that comes from, @amotl?

**Comment by amotl** [comment]
`test_log_inputs_with_duplicates_in_single_request` is flaky, see GH-11.

File: tests/test_examples.py [hunk 4]
```diff
@@ -103,3 +103,28 @@ def test_tracking_merlion(reset_database, engine: sa.Engine, tracking_store: Sql
         assert session.query(SqlExperiment).count() == 2
         assert session.query(SqlMetric).count() == 4
         assert session.query(SqlParam).count() == 5
+
+    with engine.begin() as conn:
+        conn.execute(sa.sql.text("REFRESH TABLE doc.anomalies"))
+        conn.execute(sa.sql.text("REFRESH TABLE testdrive.runs"))
```
Comment (amotl) [inline tests/test_examples.py:109]: Just calling out to `sa.text()` should work.
```suggestion
        conn.execute(sa.text("REFRESH TABLE doc.anomalies"))
        conn.execute(sa.text("REFRESH TABLE testdrive.runs"))
```

File: tests/test_examples.py [hunk 5]
```diff
@@ -103,3 +103,28 @@ def test_tracking_merlion(reset_database, engine: sa.Engine, tracking_store: Sql
...
+
+        # Anomalies that have been manually annotated
+        anomalies_manual = conn.execute(sa.sql.text("SELECT * FROM doc.anomalies WHERE experiment_id IS NULL"))
```
Comment (amotl) [inline tests/test_examples.py:112]: Dito.
```suggestion
        anomalies_manual = conn.execute(sa.text("SELECT * FROM doc.anomalies WHERE experiment_id IS NULL"))
```

File: tests/test_examples.py [hunk 6]
```diff
@@ -103,3 +103,28 @@ def test_tracking_merlion(reset_database, engine: sa.Engine, tracking_store: Sql
...
+        assert anomalies_manual.rowcount == 3
+
+        # Count anomalies that have been generated by the latest run
+        anomalies_generated = conn.execute(
+            sa.sql.text(
```
Comment (amotl) [inline tests/test_examples.py:117]: Dito.
```suggestion
            sa.text(
```

File: tests/test_examples.py (hunk 3 above)
Comment (amotl) [reply tests/test_examples.py:130]: Let's investigate on behalf of a subsequent iteration.

**Review by amotl — CHANGES_REQUESTED** [review]

**Review by amotl — APPROVED** [review]
Thank you!

**Pull request merged**